recipes/boost/all/conanfile.py

```diff
@@ -133,6 +139,8 @@ def requirements(self):
                 self.requires("xz_utils/5.2.4")
             if self.options.zstd:
                 self.requires("zstd/1.4.3")
+        if self.options.icu:
```

Comment:
probably we have to check for `without_locale` option to avoid unneeded dependency in this case

Reply:
ICU applies to `regex` and `locale`, not sure if you can apply ICU to one without the other. I really don't know how the transitivity and dependencies among Boost libraries work

Reply:
yes, seems to be check should be like:
`if not without_locale or not without_regex:`

Reply:
I've added a check. 'locale' doesn't compile without a i18n backend, but 'regex' does 🤷